### PR TITLE
tests added for fyle_utils, slack_utils and notification_messages

### PR DESCRIPTION
**Pull request by vishalpandeyvip**
Current code coverage status:
Screenshot from 2022-11-23 12-14-44

### CODE REVIEW

File: tests/messages/fixtures.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        },
+        'seq_num': 1,
+        'spent_at':'2011-08-12T20:17:46.384Z',
+        'matched_corporate_card_transactions':[
+            {
+                'corporate_card_number':'1234567891234567'
+            }
+        ]
+    }
+}
```
Comment (harshithaputtaswamy) [inline tests/messages/fixtures.py:30]: new line

File: tests/messages/test_notification_messages.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        FAKE_EXPENSE, FAKE_EXPENSE_URL = expense_data['FAKE_EXPENSE'], 'https://fakeurl.com'
+        card_expense_section_block, title_text = messages.get_expense_mandatory_receipt_missing_notification(FAKE_EXPENSE, FAKE_EXPENSE_URL)
+        display_amount = slack_utils.get_display_amount(FAKE_EXPENSE['amount'], FAKE_EXPENSE['currency'])
+        assert title_text == f':credit_card: A card expense of *{display_amount}* requires a :receipt: receipt. Please reply with a photo of your receipt in this thread!'
+        assert card_expense_section_block[0]['text']['text'] == ":credit_card: A card expense of *$100.00* requires a :receipt: receipt. Please reply with a photo of your receipt in this thread!"
+        assert card_expense_section_block[1]['fields'][0]['text'] == 'Date of Spend:\n *August 12, 2011*'
+        assert card_expense_section_block[1]['fields'][1]['text'] == "Receipt:\n :x: *Missing*"
+        assert card_expense_section_block[2]['fields'][0]['text'] == "Card No.:\n *Ending 4567 (VISA)*"
+        assert card_expense_section_block[3]['elements'][1]['value'] == "fake-id-123"
+        assert card_expense_section_block[2]['fields'][1]['text'] == "Merchant:\n *Uber*"
```
Comment (harshithaputtaswamy) [inline tests/messages/test_notification_messages.py:57]: new line

File: tests/test_fyle_utils.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        json_response = fyle_utils.attach_receipt_to_expense(FAKE_EXPENSE_ID, FAKE_RECEIPT_ID, REFRESH_TOKEN)
+        assert json_response['file_id'] == FAKE_RECEIPT_ID and json_response['id'] == FAKE_EXPENSE_ID
+
+    def test_upload_file_to_s3(self, mocker):
+        mock_response = mock.Mock(spec = Response)
+        mock_response.status_code = 200
+        mocker.patch('fyle_slack_app.libs.http.put', return_value=mock_response)
+        FAKE_UPLOAD_URL, FAKE_FILE_CONTENT , CONTENT_TYPE = 'https://fake-url.com', 'fake-file-content', 'json'
+        response = fyle_utils.upload_file_to_s3(FAKE_UPLOAD_URL, FAKE_FILE_CONTENT , CONTENT_TYPE)
+        assert response.status_code == 200
```
Comment (harshithaputtaswamy) [inline tests/test_fyle_utils.py:147]: new line

File: tests/test_slack_utils.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        except ValueError as msg:
+            assert pytest.raises(ValueError)
+
+        amount, currency = 100.5678, 'USD'
+        rounded_amount = utils.round_amount(amount, currency)
+        assert rounded_amount == round(amount + 1e-9, get_currency_precision(currency))
+
+    def test_get_display_amount(self):
+        assert "₹100,000.00" == str(utils.get_display_amount("100,000", 'INR'))
+        assert "$100,000.00" == str(utils.get_display_amount("100,000", 'USD'))
```
Comment (harshithaputtaswamy) [inline tests/test_slack_utils.py:45]: new line

**Review by Shwetabhk — APPROVED** [review]

**Pull request merged**